Switch Row and Column to flexbox

mhsmith opened the pull request:
This PR is stacked on top of #113.

The Row and Column containers have been changed from `display: grid` to `display: flex`. These containers now have the following properties:
* `align_items`
* `justify_content`
* `flex_wrap`

And their children have these properties:
* `align_self` (overrides `align_items` for an individual child)
* `flex`

The farmyard app has been updated with a couple of examples of things you couldn't do in the previous model: 

* The `number_of_honks` label is now on the same row as the slider, with the label taking only as much space as necessary, and the slider taking all the rest.
* The geese and pigs now wrap onto multiple lines (previously they overflowed the box)

Following the same pattern, I've also updated the Grid to replace the `position` property with `align_items` and `justify_items` on the container, and `align_self` and `justify_self` on the children. This allows the horizontal and vertical position to be set independently using ChoiceProperties (closes #97).

All the Grid, Row and Column child properties are still declared on Component, so that's becoming a bit cluttered. Next I plan to split them out into per-container layout classes, as we discussed the other week.

mhsmith commented:
@ntoll: You asked why we would ever NOT want `flex_wrap=True`. Here's one example: it causes it to prefer wrapping the children themselves rather than the chlidren's content. Both of these could be useful in different circumstances:

----

`flex_wrap=True`:

<img width="1006" alt="Screenshot 2024-06-11 at 21 29 50" src="https://github.com/invent-framework/invent/assets/166954/abec01ca-8693-47e9-a184-1e14b8470e5a">

----

`flex_wrap=False`:

<img width="1006" alt="Screenshot 2024-06-11 at 21 30 16" src="https://github.com/invent-framework/invent/assets/166954/0cb2d920-9de4-478f-92af-487948ee0f03">

ntoll reviewed (approved):
Awesome work here @mhsmith ... 🏴󠁧󠁢󠁳󠁣󠁴󠁿 🚀 

:shipit:

mhsmith commented:
In the end, we removed `flex_wrap` because it potentially complicates changes to drag and drop (#116), and other things  that we haven't had a chance to think through. 

We also removed `align-items` and `justify-items`, for two reasons:

* Having two ways to set alignment in the builder UI is confusing.
* We currently need every widget to have an explicit `align-self`, to override the value of that property that the Paper CSS sets on `button`, and maybe other widgets. So `align-items` wouldn't have any effect anyway.
